op-node/rollup/derive/batch_queue.go

```diff
@@ -42,24 +47,52 @@ type BatchQueue struct {
 
 	l1Blocks []eth.L1BlockRef
 
-	// batches in order of when we've first seen them, grouped by L2 timestamp
-	batches map[uint64][]*BatchWithL1InclusionBlock
+	// batches in order of when we've first seen them
+	batches []*BatchWithL1InclusionBlock
```

Comment:
@protolambda Changing to a single batch queue is required to allow overlapping span batches. In the current code, we have to scan every batches from the past because overlapping batches may not be in the queue at `nextTimestamp` 

Additionally, if I understand correctly, there might be a memory leak in the current code since we don't clean up `remaining` batches from the queue after accepting a batch. This code change also resolves this leak.